src/getLocationForJsonPath.ts

```diff
@@ -99,6 +109,38 @@ function findNodeAtPath(node: YAMLNode, path: JsonPath, closest: boolean) {
   return node;
 }
 
+function getMappings(mappings: YAMLMapping[], mergeKeys: boolean): YAMLMapping[] {
+  if (!mergeKeys) return mappings;
+
+  return mappings.reduce<YAMLMapping[]>((mergedMappings, mapping) => {
+    if (isValidNode(mapping as YAMLNode)) {
```

Comment:
I'm almost sure there's something wrong here.

1. `isValidNode` is not checking if the node is effectively valid, but just if it's not null or undefined
2. This function should probably be a type guard asserting that mapping is a `YamlNode` in case it's valid

* Either we check for null or undefined manually
* We change the `isValidNode` to `isNullOrUndefined`
* Use the type guards properly

Reply:
Yeah, you are right. This kind of acts like it was a type guard, but it's not.